src/bidiMapper/domains/context/browsingContextProcessor.ts

```diff
@@ -122,45 +150,62 @@ export class BrowsingContextProcessor {
     }
   }
 
-  async process_createContext(
-    commandData: BrowsingContext.BrowsingContextCreateCommand
-  ): Promise<BrowsingContext.BrowsingContextCreateResult> {
+  async process_PROTO_browsingContext_create(
+    commandData: BrowsingContext.PROTO.BrowsingContextCreateCommand
+  ): Promise<BrowsingContext.PROTO.BrowsingContextCreateResult> {
     const params = commandData.params;
+
     return new Promise(async (resolve) => {
-      let targetId: string;
+      const browserCdpClient = this._cdpConnection.browserClient();
+
+      const result = await browserCdpClient.Target.createTarget({
+        url: 'about:blank',
+        newWindow: params.type === 'window',
+      });
+
+      const targetId = result.targetId;
+
+      const existingContext = await this._tryGetContext(targetId);
+      if (existingContext) {
+        resolve(existingContext.toBidi());
```

Comment:
I think awaiting here might introduce a potential race condition. Once Target.createTarget returns, we need to register the onAttachToTarget handler synchronously, or there's a chance the event will come and we'll miss it.

Reply:
Agree. Added synchronous check, if the context is already known, in which case `attachedToTarget` is not needed.

Reply:
Thank you, that makes more sense. I think this is safe then. Even though `await this._tryGetContext(targetId)` will yield to the runtime, any `attachedToTarget` event should be handled by `_handleAttachedToTargetEvent` and the context will be put in the map before `_tryGetContext` actually executes.